Src/main.cpp

```diff
@@ -161,20 +161,56 @@ void special(int key, int x, int y)
 {
     switch (key)
     {
+    case GLUT_KEY_F1:
+        h89.keypress('S' | 0x80);
+        break;
+
+    case GLUT_KEY_F2:
+        h89.keypress('T' | 0x80);
+        break;
+
+    case GLUT_KEY_F3:
+        h89.keypress('U' | 0x80);
+        break;
+
+    case GLUT_KEY_F4:
+        h89.keypress('V' | 0x80);
+        break;
+
+    case GLUT_KEY_F5:
+        h89.keypress('W' | 0x80);
+        break;
+
+    case GLUT_KEY_F6:
+        h89.keypress('P' | 0x80);
+        break;
+
+    case GLUT_KEY_F7:
+        h89.keypress('Q' | 0x80);
+        break;
+
+    case GLUT_KEY_F8:
+        h89.keypress('R' | 0x80);
+        break;
+
+    case GLUT_KEY_HOME:
+        h89.keypress('H' | 0x80);
+        break;
+
     case GLUT_KEY_UP:
-        h89.keypress('8');
+        h89.keypress('A' | 0x80);
```

Comment:
What I was thinking. On a modern keyboard even thought there is both the arrow keys and the number pad, the number pad will send arrows,pageUp/Down, etc, if the num lock is turned off. It also has an 'alternate mode', which sends different ESC sequences. See page 22 in the H19 manual(http://koyado.com/Heathkit/H8-Z19_files/Heath%20Z19%20Terminal%20Operating%20Manual.pdf) - there are four potential states. The eventual plan, is to have they numpad send the right values based on the current shifted and alternate modes set by ESC sequences and/or dipswitch settings on the terminal board.